filepattern-generator-plugin/src/main.py

```diff
@@ -0,0 +1,201 @@
+        else:
+            self.fp = filepattern.FilePattern(self.inpDir, self.pattern,var_order='rxytpc')
+            self.fileslist = [file[0] for file in self.fp(group_by=self.groupBy)]
+
+        
+    def batch_chunker(self):
+        """This function uses List of image files and chunkSize and creates iterator
+        Args:
+            fileslist: List of Image file names.
+            chunkSize (str): Number of images to generate collective filepattern
+
+        Returns:
+            Iterator
+        """
+        batch_iterator = iter(self.fileslist)
+        while True:
+            chunk = []
+            try:
+                for _ in range(self.chunkSize):
+                    chunk.append(next(batch_iterator))
+                yield chunk
+            except StopIteration:
+                if chunk:
+                    yield chunk
+                return
+
+
+    def pattern_generator(self):
+        """This function iterates over batches of image files and returns filepatterns for each batch and batch sizes
+        Args:
+            iterator: calling batch_chunker function
+            pattern (str): Filepattern to parse image files
+
+        Returns:
+            pandas DataFrame
+        
+        """
+        pf = self.batch_chunker() 
+        df = []
+        batch_size=[]
+        for _, batch in enumerate(pf):
+            batch_files = []
+            for b in batch:
+                batch_files.append(b)
+            if self.pattern is None:
+                pattern_regex = filepattern.infer_pattern(batch_files)
+            else:
+                pattern_regex = self.fp.output_name(batch_files)
+            df.append(pattern_regex)
+            batch_size.append(len(batch))
+
+        prf = pd.DataFrame(list(zip(df, batch_size)), columns=['FilePattern', 'Batch_Size'])
+        return prf
+
+    def saving_generator_outputs(self):
+        """Saving Outputs to CSV/Feather file format
+        Args:
+            DataFrame: calling pattern_generator function
+
+        Returns:
+            CSV/Feather format file
+            """    
+        prf = self.pattern_generator()
+        os.chdir(self.outDir)
+        if self.outFormat == 'csv':
+            prf.to_csv('pattern_generator.csv', index=False)
+        elif self.outFormat == 'feather':
+            pyarrow.feather.write_feather(prf, "pattern_generator.feather")
+        else:
+            prf.to_csv('pattern_generator.csv', index=False)
+            pyarrow.feather.write_feather(prf, "pattern_generator.feather")
+     
+        return
+
+  
+def main(inpDir:Path,
+         outDir:Path,
+         pattern:str,
+         chunkSize:int,
+         groupBy:str,
+         outFormat:str):
+    starttime= time.time()
+
+    logger.info(f'Start parsing Image Filenames') 
+    assert inpDir.exists(), logger.info('Input directory does not exist')
+    fg = Filepattern_Generator(inpDir,outDir, pattern,chunkSize,groupBy,outFormat)
+    fg.saving_generator_outputs()
```

Comment:
It would be nice to see a subset of the file pattern outputs logged.  Not all of them but a small subset.  

I'm thinking that a user may want to verify the file pattern worked by looking in the logs.  Right now, you would require someone to view the csv collection to see if this actually worked.

You should not log the entire output but maybe a small subset of the output collection could be useful.  

I would defer to @Nicholas-Schaub if he thinks this is a dumb idea.

Reply:
I have added to show the subset of outputs by default first 30 in the logs.